scripts/MultiPlotting/multi_plotting_context.py

```diff
@@ -34,13 +34,11 @@ def addLine(self, subplotName, workspace, specNum):
         try:
             if isinstance(workspace, str):
                 ws = mantid.AnalysisDataService.retrieve(workspace)
-                self.subplots[subplotName].addLine(ws, specNum)
-
             elif len(workspace) > 1:
-                self.subplots[subplotName].addLine(
-                    workspace.OutputWorkspace, specNum)
+                ws = workspace.OutputWorkspace
             else:
-                self.subplots[subplotName].addLine(workspace, specNum)
+                ws = workspace
+            self.subplots[subplotName].addLine(ws, specNum)
```

Comment:
ws has the risk of being out of scope as it is not defined outside of the if statements. You would need to have `ws = None` or something similar before the if statement.